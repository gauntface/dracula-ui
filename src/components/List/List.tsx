import cx from 'classnames/dedupe'
import React, { HTMLAttributes } from 'react'
import { ColorMap } from '../../base/colors'
import {
  marginMixin,
  MarginMixin,
  paddingMixin,
  PaddingMixin
} from '../../base/spacing'

export const listColors: Partial<ColorMap> = {
  white: 'drac-list-white',
  cyan: 'drac-list-cyan',
  green: 'drac-list-green',
  orange: 'drac-list-orange',
  pink: 'drac-list-pink',
  purple: 'drac-list-purple',
  red: 'drac-list-red',
  yellow: 'drac-list-yellow'
}

export const listVariants = {
  unordered: 'drac-list-unordered',
  ordered: 'drac-list-ordered',
  none: 'drac-list-none'
}

/**
 * List Props
 */
export interface ListProps
  extends HTMLAttributes<HTMLOListElement>,
    PaddingMixin,
    MarginMixin {
  /** The Dracula UI color for the List. */
  color: keyof typeof listColors

  /**
   * The variation to be used for the List element.
   * `unordered` -> Display list in dashes as an `<ul>`
   * `ordered` -> Display list in numbers as an `<ol>`
   */
  variant?: keyof typeof listVariants
}

/**
 * Lists are horizontal lines used to separate semantic blocks of
 * content or UI patterns.
 */
export const List: React.FC<ListProps> = (props: ListProps) => {
  const finalProps = {
    ...props,
    className: cx(
      'drac-list',
      props.variant && listVariants[props.variant],
      props.color && listColors[props.color],
      ...paddingMixin(props),
      ...marginMixin(props)
    )
  }

  if (props.variant === 'unordered') {
    return <ul {...finalProps} />
  }

  return <ol {...finalProps} />
}

List.displayName = 'List'
