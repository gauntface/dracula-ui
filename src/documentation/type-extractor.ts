import * as ts from 'typescript'

export function extractWithComment(
  fileNames: string[],
  options: ts.CompilerOptions
): void {
  const program = ts.createProgram(fileNames, options)
  const checker: ts.TypeChecker = program.getTypeChecker()

  for (const sourceFile of program.getSourceFiles()) {
    if (!sourceFile.isDeclarationFile) {
      ts.forEachChild(sourceFile, visit)
    }
  }

  function visit(node: ts.Node) {
    const count = node.getChildCount()

    if (count > 0) {
      ts.forEachChild(node, visit)
    }

    if (ts.isInterfaceDeclaration(node) && node.name) {
      const symbol = checker.getSymbolAtLocation(node.name)
      if (symbol) {
        return serializeSymbol(symbol)
      }
    }
  }

  function serializeSymbol(symbol: ts.Symbol) {
    const s = {
      name: symbol.getName(),
      comment: ts.displayPartsToString(symbol.getDocumentationComment(checker)),
      type: checker.typeToString(
        checker.getTypeOfSymbolAtLocation(symbol, symbol.valueDeclaration!)
      )
    }
    console.log(s)
    return s
  }
}

console.log(extractWithComment(['src/components/Avatar/Avatar.tsx'], {}))
