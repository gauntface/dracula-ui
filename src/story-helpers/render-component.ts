import testRenderer from 'react-test-renderer'

export type ComponentExample = {
  code: ReturnType<testRenderer.ReactTestRenderer['toJSON']>
  docs: string
  title: string
}

export type SnapshotBuilder = {
  comp: () => React.ReactElement
  docs: string
  title: string
}

export function buildSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): SnapshotBuilder {
  return {
    title,
    comp,
    docs: docs.trim()
  }
}

export function renderSnapshot(
  title: string,
  comp: React.ReactElement,
  docs: string = ''
): ComponentExample {
  const rendered = testRenderer.create(comp)
  const code = rendered.toJSON()
  return {
    title,
    code,
    docs
  }
}
