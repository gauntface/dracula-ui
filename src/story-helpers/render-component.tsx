import React from 'react'
import Enzyme, { mount,  shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import prettier from 'prettier'
import toJson from "enzyme-to-json";

Enzyme.configure({ adapter: new Adapter() })

export type ComponentExample = {
  html: string
  docs: string
  title: string
  react?: any
}

export type SnapshotBuilder = {
  comp: () => React.ReactElement
  docs: string
  title: string
}

export function buildSnapshot(
  title: string,
  comp: () => React.ReactElement,
  docs: string = ''
): SnapshotBuilder {
  return {
    title,
    comp,
    docs: docs.trim()
  }
}

export function renderSnapshot(
  title: string,
  comp: React.ReactElement,
  docs: string = ''
): ComponentExample {

  
  const html = mount(comp).html()
  const react = toJson(shallow(<div>{comp}</div>))

  const prettyHTML = prettier.format(html, {
    parser: 'html'
  })

  return {
    title,
    html: prettyHTML,
    react,
    docs
  }
}
