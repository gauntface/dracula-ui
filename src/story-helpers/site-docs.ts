import {
  renderSnapshot,
  SnapshotBuilder
} from '@/story-helpers/render-component'
import React from 'react'

// import * as serializer from 'jest-snapshot-serializer-raw'

interface Documentation {
  basicUsage: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder>
}

export function siteDocs<T>(
  componentClass: React.FunctionComponent<T>,
  documentation: Documentation
) {
  describe(`Site: ${componentClass.displayName ?? componentClass.name}`, () => {
    test('Basic Usage', () => {
      const variation = documentation.basicUsage()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp(),
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()
    })

    const variations = documentation.variations()

    variations.forEach(variation => {
      test(variation.title, () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp(),
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()
      })
    })
  })
}
