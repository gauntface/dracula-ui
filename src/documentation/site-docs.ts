import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from '../story-helpers/render-component'
import React from 'react'
import fs from 'fs-extra'
import { pretty } from '../story-helpers/pretty'
import { getDocGen } from '../documentation/docgen/doc-generator'
import { toDSP } from '../documentation/dsp/component-generator'

interface Documentation {
  basicUsage: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder>
}

export function siteDocs<T>(
  componentClass: React.FunctionComponent<T>,
  documentation: Documentation
) {
  const name = componentClass.displayName ?? componentClass.name

  describe(`Site: ${name}`, () => {
    let examples: Record<string, ComponentExample> = {}

    afterAll(async () => {
      const docGen = getDocGen(name)
      const dsp = toDSP(name, examples, docGen)
      const dspPath = `${process.cwd()}/dsp/data/components/${name}.json`
      await fs.writeFile(dspPath, pretty(JSON.stringify(dsp), 'json'))
    })

    test('Basic Usage', () => {
      const variation = documentation.basicUsage()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()
      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations = documentation.variations()

    variations.forEach((variation) => {
      test(variation.title, () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
