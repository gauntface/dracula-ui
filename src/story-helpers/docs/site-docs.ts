import {
  ComponentExample,
  renderSnapshot,
  SnapshotBuilder
} from '@/story-helpers/render-component'
import React from 'react'
import fs from 'fs'
import { pretty } from '@/story-helpers/pretty'

interface Documentation {
  basicUsage: () => SnapshotBuilder
  variations: () => Array<SnapshotBuilder>
}

export function siteDocs<T>(
  componentClass: React.FunctionComponent<T>,
  documentation: Documentation
) {
  const name = componentClass.displayName ?? componentClass.name

  describe(`Site: ${name}`, () => {
    let examples: Record<string, ComponentExample> = {}

    afterAll(() => {
      const path = `${process.cwd()}/examples/${name}.json`
      fs.writeFileSync(path, pretty(JSON.stringify(examples), 'json'))
    })

    test('Basic Usage', () => {
      const variation = documentation.basicUsage()
      const snapshot = renderSnapshot(
        variation.title,
        variation.comp,
        variation.docs
      )
      expect(snapshot).toMatchSnapshot()
      examples = {
        ...examples,
        basic: snapshot
      }
    })

    const variations = documentation.variations()

    variations.forEach((variation) => {
      test(variation.title, () => {
        const snapshot = renderSnapshot(
          variation.title,
          variation.comp,
          variation.docs
        )
        expect(snapshot).toMatchSnapshot()

        examples = {
          ...examples,
          [variation.title]: snapshot
        }
      })
    })
  })
}
