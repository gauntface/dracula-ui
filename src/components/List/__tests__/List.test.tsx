import { List } from '@/components/List/List'
import { docs } from '@/documentation/site-docs'
import {
  ListNormal,
  ListColors,
  ListVariants
} from '@/components/List/__stories__/List.stories'
import { snapshot } from '../../../documentation/render-component'

docs(List, {
  basic() {
    return snapshot('Usage', ListNormal)
  },
  variations() {
    return [snapshot('Colors', ListColors), snapshot('Variants', ListVariants)]
  }
})
