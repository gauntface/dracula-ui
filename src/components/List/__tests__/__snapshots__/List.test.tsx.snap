// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Site: List Basic Usage 1`] = `
Object {
  "docs": "",
  "html": "<div style=\\"display: flex; flex-direction: column; padding: 3rem\\">
  <ol color=\\"purple\\" class=\\"drac-list drac-list-purple\\">
    <li class=\\"drac-text drac-text-white\\">Blade</li>
    <li class=\\"drac-text drac-text-white\\">Buffy</li>
    <li class=\\"drac-text drac-text-white\\">Morbius</li>
  </ol>
</div>
",
  "react": "<div style={{ display: \\"flex\\", flexDirection: \\"column\\", padding: \\"3rem\\" }}>
  <List color=\\"purple\\">
    <li className=\\"drac-text drac-text-white\\">Blade</li>
    <li className=\\"drac-text drac-text-white\\">Buffy</li>
    <li className=\\"drac-text drac-text-white\\">Morbius</li>
  </List>
</div>;
",
  "title": "Usage",
}
`;

exports[`Site: List Colors 1`] = `
Object {
  "docs": "",
  "html": "<div style=\\"display: flex; flex-direction: column; padding: 3rem\\">
  <div>
    <ol
      variant=\\"ordered\\"
      color=\\"white\\"
      class=\\"drac-list drac-list-ordered drac-list-white\\"
    >
      <li class=\\"drac-text drac-text-white\\">Blade</li>
      <li class=\\"drac-text drac-text-white\\">Buffy</li>
      <li class=\\"drac-text drac-text-white\\">Morbius</li>
    </ol>
  </div>
  <div>
    <ol
      variant=\\"ordered\\"
      color=\\"cyan\\"
      class=\\"drac-list drac-list-ordered drac-list-cyan\\"
    >
      <li class=\\"drac-text drac-text-white\\">Blade</li>
      <li class=\\"drac-text drac-text-white\\">Buffy</li>
      <li class=\\"drac-text drac-text-white\\">Morbius</li>
    </ol>
  </div>
  <div>
    <ol
      variant=\\"ordered\\"
      color=\\"green\\"
      class=\\"drac-list drac-list-ordered drac-list-green\\"
    >
      <li class=\\"drac-text drac-text-white\\">Blade</li>
      <li class=\\"drac-text drac-text-white\\">Buffy</li>
      <li class=\\"drac-text drac-text-white\\">Morbius</li>
    </ol>
  </div>
  <div>
    <ol
      variant=\\"ordered\\"
      color=\\"orange\\"
      class=\\"drac-list drac-list-ordered drac-list-orange\\"
    >
      <li class=\\"drac-text drac-text-white\\">Blade</li>
      <li class=\\"drac-text drac-text-white\\">Buffy</li>
      <li class=\\"drac-text drac-text-white\\">Morbius</li>
    </ol>
  </div>
  <div>
    <ol
      variant=\\"ordered\\"
      color=\\"pink\\"
      class=\\"drac-list drac-list-ordered drac-list-pink\\"
    >
      <li class=\\"drac-text drac-text-white\\">Blade</li>
      <li class=\\"drac-text drac-text-white\\">Buffy</li>
      <li class=\\"drac-text drac-text-white\\">Morbius</li>
    </ol>
  </div>
  <div>
    <ol
      variant=\\"ordered\\"
      color=\\"purple\\"
      class=\\"drac-list drac-list-ordered drac-list-purple\\"
    >
      <li class=\\"drac-text drac-text-white\\">Blade</li>
      <li class=\\"drac-text drac-text-white\\">Buffy</li>
      <li class=\\"drac-text drac-text-white\\">Morbius</li>
    </ol>
  </div>
  <div>
    <ol
      variant=\\"ordered\\"
      color=\\"red\\"
      class=\\"drac-list drac-list-ordered drac-list-red\\"
    >
      <li class=\\"drac-text drac-text-white\\">Blade</li>
      <li class=\\"drac-text drac-text-white\\">Buffy</li>
      <li class=\\"drac-text drac-text-white\\">Morbius</li>
    </ol>
  </div>
  <div>
    <ol
      variant=\\"ordered\\"
      color=\\"yellow\\"
      class=\\"drac-list drac-list-ordered drac-list-yellow\\"
    >
      <li class=\\"drac-text drac-text-white\\">Blade</li>
      <li class=\\"drac-text drac-text-white\\">Buffy</li>
      <li class=\\"drac-text drac-text-white\\">Morbius</li>
    </ol>
  </div>
  <div>
    <ol variant=\\"ordered\\" color=\\"animated\\" class=\\"drac-list drac-list-ordered\\">
      <li class=\\"drac-text drac-text-white\\">Blade</li>
      <li class=\\"drac-text drac-text-white\\">Buffy</li>
      <li class=\\"drac-text drac-text-white\\">Morbius</li>
    </ol>
  </div>
  <div>
    <ol variant=\\"ordered\\" color=\\"cyanGreen\\" class=\\"drac-list drac-list-ordered\\">
      <li class=\\"drac-text drac-text-white\\">Blade</li>
      <li class=\\"drac-text drac-text-white\\">Buffy</li>
      <li class=\\"drac-text drac-text-white\\">Morbius</li>
    </ol>
  </div>
  <div>
    <ol
      variant=\\"ordered\\"
      color=\\"pinkPurple\\"
      class=\\"drac-list drac-list-ordered\\"
    >
      <li class=\\"drac-text drac-text-white\\">Blade</li>
      <li class=\\"drac-text drac-text-white\\">Buffy</li>
      <li class=\\"drac-text drac-text-white\\">Morbius</li>
    </ol>
  </div>
  <div>
    <ol
      variant=\\"ordered\\"
      color=\\"purpleCyan\\"
      class=\\"drac-list drac-list-ordered\\"
    >
      <li class=\\"drac-text drac-text-white\\">Blade</li>
      <li class=\\"drac-text drac-text-white\\">Buffy</li>
      <li class=\\"drac-text drac-text-white\\">Morbius</li>
    </ol>
  </div>
  <div>
    <ol
      variant=\\"ordered\\"
      color=\\"yellowPink\\"
      class=\\"drac-list drac-list-ordered\\"
    >
      <li class=\\"drac-text drac-text-white\\">Blade</li>
      <li class=\\"drac-text drac-text-white\\">Buffy</li>
      <li class=\\"drac-text drac-text-white\\">Morbius</li>
    </ol>
  </div>
</div>
",
  "react": "<div style={{ display: \\"flex\\", flexDirection: \\"column\\", padding: \\"3rem\\" }}>
  <div>
    <List variant=\\"ordered\\" color=\\"white\\">
      <li className=\\"drac-text drac-text-white\\">Blade</li>
      <li className=\\"drac-text drac-text-white\\">Buffy</li>
      <li className=\\"drac-text drac-text-white\\">Morbius</li>
    </List>
  </div>
  <div>
    <List variant=\\"ordered\\" color=\\"cyan\\">
      <li className=\\"drac-text drac-text-white\\">Blade</li>
      <li className=\\"drac-text drac-text-white\\">Buffy</li>
      <li className=\\"drac-text drac-text-white\\">Morbius</li>
    </List>
  </div>
  <div>
    <List variant=\\"ordered\\" color=\\"green\\">
      <li className=\\"drac-text drac-text-white\\">Blade</li>
      <li className=\\"drac-text drac-text-white\\">Buffy</li>
      <li className=\\"drac-text drac-text-white\\">Morbius</li>
    </List>
  </div>
  <div>
    <List variant=\\"ordered\\" color=\\"orange\\">
      <li className=\\"drac-text drac-text-white\\">Blade</li>
      <li className=\\"drac-text drac-text-white\\">Buffy</li>
      <li className=\\"drac-text drac-text-white\\">Morbius</li>
    </List>
  </div>
  <div>
    <List variant=\\"ordered\\" color=\\"pink\\">
      <li className=\\"drac-text drac-text-white\\">Blade</li>
      <li className=\\"drac-text drac-text-white\\">Buffy</li>
      <li className=\\"drac-text drac-text-white\\">Morbius</li>
    </List>
  </div>
  <div>
    <List variant=\\"ordered\\" color=\\"purple\\">
      <li className=\\"drac-text drac-text-white\\">Blade</li>
      <li className=\\"drac-text drac-text-white\\">Buffy</li>
      <li className=\\"drac-text drac-text-white\\">Morbius</li>
    </List>
  </div>
  <div>
    <List variant=\\"ordered\\" color=\\"red\\">
      <li className=\\"drac-text drac-text-white\\">Blade</li>
      <li className=\\"drac-text drac-text-white\\">Buffy</li>
      <li className=\\"drac-text drac-text-white\\">Morbius</li>
    </List>
  </div>
  <div>
    <List variant=\\"ordered\\" color=\\"yellow\\">
      <li className=\\"drac-text drac-text-white\\">Blade</li>
      <li className=\\"drac-text drac-text-white\\">Buffy</li>
      <li className=\\"drac-text drac-text-white\\">Morbius</li>
    </List>
  </div>
  <div>
    <List variant=\\"ordered\\" color=\\"animated\\">
      <li className=\\"drac-text drac-text-white\\">Blade</li>
      <li className=\\"drac-text drac-text-white\\">Buffy</li>
      <li className=\\"drac-text drac-text-white\\">Morbius</li>
    </List>
  </div>
  <div>
    <List variant=\\"ordered\\" color=\\"cyanGreen\\">
      <li className=\\"drac-text drac-text-white\\">Blade</li>
      <li className=\\"drac-text drac-text-white\\">Buffy</li>
      <li className=\\"drac-text drac-text-white\\">Morbius</li>
    </List>
  </div>
  <div>
    <List variant=\\"ordered\\" color=\\"pinkPurple\\">
      <li className=\\"drac-text drac-text-white\\">Blade</li>
      <li className=\\"drac-text drac-text-white\\">Buffy</li>
      <li className=\\"drac-text drac-text-white\\">Morbius</li>
    </List>
  </div>
  <div>
    <List variant=\\"ordered\\" color=\\"purpleCyan\\">
      <li className=\\"drac-text drac-text-white\\">Blade</li>
      <li className=\\"drac-text drac-text-white\\">Buffy</li>
      <li className=\\"drac-text drac-text-white\\">Morbius</li>
    </List>
  </div>
  <div>
    <List variant=\\"ordered\\" color=\\"yellowPink\\">
      <li className=\\"drac-text drac-text-white\\">Blade</li>
      <li className=\\"drac-text drac-text-white\\">Buffy</li>
      <li className=\\"drac-text drac-text-white\\">Morbius</li>
    </List>
  </div>
</div>;
",
  "title": "Colors",
}
`;

exports[`Site: List Variants 1`] = `
Object {
  "docs": "",
  "html": "<div style=\\"display: flex; flex-direction: column; padding: 3rem\\">
  <div>
    <ul
      variant=\\"unordered\\"
      color=\\"purple\\"
      class=\\"drac-list drac-list-unordered drac-list-purple\\"
    >
      <li class=\\"drac-text drac-text-white\\">This is the first item</li>
      <li class=\\"drac-text drac-text-white\\">This is the second item</li>
      <li class=\\"drac-text drac-text-white\\">This is the third item</li>
      <li class=\\"drac-text drac-text-white\\">This is the fourth item</li>
      <li class=\\"drac-text drac-text-white\\">This is the fifth item</li>
      <li class=\\"drac-text drac-text-white\\">This is the sixth item</li>
      <li class=\\"drac-text drac-text-white\\">This is the seventh item</li>
      <li class=\\"drac-text drac-text-white\\">This is the eighth item</li>
      <li class=\\"drac-text drac-text-white\\">This is the ninth item</li>
      <li class=\\"drac-text drac-text-white\\">
        This is the tenth item (showing how numbers align and wrapping looks by
        default)
      </li>
    </ul>
  </div>
  <div>
    <ol
      variant=\\"ordered\\"
      color=\\"purple\\"
      class=\\"drac-list drac-list-ordered drac-list-purple\\"
    >
      <li class=\\"drac-text drac-text-white\\">This is the first item</li>
      <li class=\\"drac-text drac-text-white\\">This is the second item</li>
      <li class=\\"drac-text drac-text-white\\">This is the third item</li>
      <li class=\\"drac-text drac-text-white\\">This is the fourth item</li>
      <li class=\\"drac-text drac-text-white\\">This is the fifth item</li>
      <li class=\\"drac-text drac-text-white\\">This is the sixth item</li>
      <li class=\\"drac-text drac-text-white\\">This is the seventh item</li>
      <li class=\\"drac-text drac-text-white\\">This is the eighth item</li>
      <li class=\\"drac-text drac-text-white\\">This is the ninth item</li>
      <li class=\\"drac-text drac-text-white\\">
        This is the tenth item (showing how numbers align and wrapping looks by
        default)
      </li>
    </ol>
  </div>
  <div>
    <ol
      variant=\\"none\\"
      color=\\"purple\\"
      class=\\"drac-list drac-list-none drac-list-purple\\"
    >
      <li class=\\"drac-text drac-text-white\\">This is the first item</li>
      <li class=\\"drac-text drac-text-white\\">This is the second item</li>
      <li class=\\"drac-text drac-text-white\\">This is the third item</li>
      <li class=\\"drac-text drac-text-white\\">This is the fourth item</li>
      <li class=\\"drac-text drac-text-white\\">This is the fifth item</li>
      <li class=\\"drac-text drac-text-white\\">This is the sixth item</li>
      <li class=\\"drac-text drac-text-white\\">This is the seventh item</li>
      <li class=\\"drac-text drac-text-white\\">This is the eighth item</li>
      <li class=\\"drac-text drac-text-white\\">This is the ninth item</li>
      <li class=\\"drac-text drac-text-white\\">
        This is the tenth item (showing how numbers align and wrapping looks by
        default)
      </li>
    </ol>
  </div>
</div>
",
  "react": "<div style={{ display: \\"flex\\", flexDirection: \\"column\\", padding: \\"3rem\\" }}>
  <div>
    <List variant=\\"unordered\\" color=\\"purple\\">
      <li className=\\"drac-text drac-text-white\\">This is the first item</li>
      <li className=\\"drac-text drac-text-white\\">This is the second item</li>
      <li className=\\"drac-text drac-text-white\\">This is the third item</li>
      <li className=\\"drac-text drac-text-white\\">This is the fourth item</li>
      <li className=\\"drac-text drac-text-white\\">This is the fifth item</li>
      <li className=\\"drac-text drac-text-white\\">This is the sixth item</li>
      <li className=\\"drac-text drac-text-white\\">This is the seventh item</li>
      <li className=\\"drac-text drac-text-white\\">This is the eighth item</li>
      <li className=\\"drac-text drac-text-white\\">This is the ninth item</li>
      <li className=\\"drac-text drac-text-white\\">
        This is the tenth item (showing how numbers align and wrapping looks by
        default)
      </li>
    </List>
  </div>
  <div>
    <List variant=\\"ordered\\" color=\\"purple\\">
      <li className=\\"drac-text drac-text-white\\">This is the first item</li>
      <li className=\\"drac-text drac-text-white\\">This is the second item</li>
      <li className=\\"drac-text drac-text-white\\">This is the third item</li>
      <li className=\\"drac-text drac-text-white\\">This is the fourth item</li>
      <li className=\\"drac-text drac-text-white\\">This is the fifth item</li>
      <li className=\\"drac-text drac-text-white\\">This is the sixth item</li>
      <li className=\\"drac-text drac-text-white\\">This is the seventh item</li>
      <li className=\\"drac-text drac-text-white\\">This is the eighth item</li>
      <li className=\\"drac-text drac-text-white\\">This is the ninth item</li>
      <li className=\\"drac-text drac-text-white\\">
        This is the tenth item (showing how numbers align and wrapping looks by
        default)
      </li>
    </List>
  </div>
  <div>
    <List variant=\\"none\\" color=\\"purple\\">
      <li className=\\"drac-text drac-text-white\\">This is the first item</li>
      <li className=\\"drac-text drac-text-white\\">This is the second item</li>
      <li className=\\"drac-text drac-text-white\\">This is the third item</li>
      <li className=\\"drac-text drac-text-white\\">This is the fourth item</li>
      <li className=\\"drac-text drac-text-white\\">This is the fifth item</li>
      <li className=\\"drac-text drac-text-white\\">This is the sixth item</li>
      <li className=\\"drac-text drac-text-white\\">This is the seventh item</li>
      <li className=\\"drac-text drac-text-white\\">This is the eighth item</li>
      <li className=\\"drac-text drac-text-white\\">This is the ninth item</li>
      <li className=\\"drac-text drac-text-white\\">
        This is the tenth item (showing how numbers align and wrapping looks by
        default)
      </li>
    </List>
  </div>
</div>;
",
  "title": "Variants",
}
`;
